packages/langium-cli/src/parser-validation.ts

```diff
@@ -23,20 +25,61 @@ export function validateParser(grammar: Grammar, config: LangiumConfig): Error |
     };
     const generatedModule: Module<LangiumServices, LangiumGeneratedServices> = {
         Grammar: () => grammar,
-        LanguageMetaData: unavailable,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        LanguageMetaData: () => grammarConfigMap.get(grammar) as any,
         parser: {
             ParserConfig: () => parserConfig
         }
     };
     const shared = inject(createDefaultSharedModule(), generatedSharedModule);
     const services = inject(createDefaultModule({ shared }), generatedModule);
+
+    let parser: LangiumParser | undefined;
     try {
-        createLangiumParser(services);
+        parser = prepareLangiumParser(services);
+        // The finalization step invokes parser validation, which can lead to thrown errors
+        parser.finalize();
         return undefined;
     } catch (err) {
+        if (parser && parser.definitionErrors.length > 0) {
+            // Construct a message with tracing information
+            let message = 'Parser definition errors detected:';
+            for (const defError of parser.definitionErrors) {
+                message += '\n-------------------------------\n';
+                if (defError.ruleName) {
+                    const rule = findRule(defError.ruleName, grammar, documents);
```

Comment:
Due to #337, Chevrotain rule names have a `:RULE` suffix to prevent clashing with terminal names. The changes don't reflect that. I just noticed that this looks kind of silly in the Chevrotain error messages, so perhaps we should think about changing that.

Reply:
Yes, it would be great to see the plain rule names in the error messages.

CC @Pluralia

Reply:
Thank you, I've reopened [the issue](https://github.com/langium/langium/issues/294) and make a new PR with another solution.

Reply:
Ho to continue? Wait for the other solution or merge this (it shouldn't hurt)?